### PR TITLE
Removing undocumented functionality that hooks into Caps Lock key and slows down mouse movement.

### LINKED ISSUE #6: Remove "caps lock logic" and establish baseline rules for functionality of application
There's some functionality in MouseUnSnag that is undocumented and doesn't really adhere to what I'd expect to be the core thesis in terms of functionality for this application (i.e. preventing the mouse from getting stuck in corners). The only caveat to this should ever be Dale's original "wrap" functionality (just a classic carryover).

I noticed this in the code and found it pretty surprising and I'd like to start a discussion to determine if this should remain present: Essentially, if you hold down caps lock, the mouse slows down. Of course, CAPS is still activated (or deactivated) when pressing, as you'd expect. It's just... weird to me.

https://github.com/MouseUnSnag/MouseUnSnag/blob/master/MouseUnSnag/MouseHookHandler.cs#L46-L63

### PR DESCRIPTION
**Pull request by patricknelson**
Fixes #6. This eviscerates undocumented functionality that hooks into <kbd>Caps Lock</kbd> key and slows down mouse movement. It also apparently fixes the same bug that PR #5 attempts to address (i.e. the bug where clicking "Exit" didn't terminate `MouseUnSnag.exe`).

Turns out the root cause there was due to a separate thread still running in the background. With this thread removed, `Application.Exit()` terminates the main process as you'd expect. Also... no weird mouse slowdown side effects when depressing the <kbd>Caps Lock</kbd> key, either.

### CODE REVIEW

**Comment by patricknelson** [comment]
p.s. It seems that there's a bunch of other stuff left over as well (which I haven't removed yet) which appear to only be used by the  the <kbd>Caps Lock</kbd> functionality that was incorporated earlier that this PR attempts to remove. This includes the `Win32Keyboard` and `KeyPressedStates` classes which serve no purpose anymore without this undocumented feature.

**Comment by patricknelson** [comment]
@GrantMoyer adding you as a reviewer here; you're probably much more familiar with C# than I am. I also stripped out a few extra classes which weren't being used at all since we're not watching the keyboard in this app.

Personally I'm strongly in favor going this route, removing the <kbd>Caps Lock</kbd>-based "mouse slowdown" functionality (as it seems well outside of the scope of what I think this MouseUnSnag app should do). The benefit is that this would also address the same bug addressed in #5 and fixing the "Exit" bug where the `MouseUnSnag.exe` process doesn't actually close. If this is merged, we should close out that other PR too, of course.

**Comment by patricknelson** [comment]
That reminds me: What’s the git workflow here? I’m assuming

- `master` branch is blessed (some use it as a dev or bleeding edge branch)
- All feature branches are branched off of `master`
- A `dev` branch may exist as a bleeding edge development branch (for integration)

Only minor issue with that last one is just that the `dev` branch would have to be reverted or reset from time to time in case features don’t ever make it. In my projects, the `dev` branch doesn’t ever make it into `master` as it’s purely used for experimentation and ensuring feature branches work together. Thoughts?

**Comment by patricknelson** [comment]
Also, regarding the functionality itself, my approach would be:

- Remove it from `master` (the branch upon which we base our releases). My goal is to hopefully not only fix this `Exit` bug but to also get folks using the settings retention feature (which is super useful, see #11). 
- At _most_ convert it to a feature branch (i.e. branches off `master` and _only_ contains that additional feature). Only since we don't currently have a process for `dev` branch feature integration/testing (yet).
 
However, IMHO, we should be decisive and include it (and fix it) or not. I, of course, vote to just remove it completely and allow others to pull it and use it as they see fit in separate apps.

My reasoning is:

1. It's hard to justify since it doesn't stick to the core functionality of the app. That is: Get the mouse unstuck from corners or screen edges, particularly between monitors. That's it (hence the name). Heck, even by that logic, the "wrapping" functionality (frequently requested for removal) doesn't meet this criteria... but that's a separate topic.
2. It's not configurable, so that would have to be added. Why <kbd>Caps Lock</kbd> and not <kbd>Left Shift</kbd>? Or maybe <kbd>Scroll Lock</kbd> (etc). Basically it should then have additional configuration where you then can plug in which key should be used as a global modifier.

I only say all this because from [the original superuser.com post](https://superuser.com/questions/947817/how-to-disable-sticky-corners-in-windows-10) from 6yrs ago, there's a lot of interest in this and with that wide of a variety of potential users, it's best to build the app to be as small and simple as possible to solve this problem and to solve it well in the most compatible way possible.

**Comment by patricknelson** [comment]
p.s. @GrantMoyer there's also 431be82 ("Refactor Win32 stuff") which introduced some keyboard-specific functionality as well relating to <kbd>Caps Lock</kbd>, but (as I feared regarding just reversing commits) some of these commits overlap a bit with non-<kbd>Caps Lock</kbd> stuff, so the best solution may be to just manually remove all the keyboard related-functionality directly (referencing these commits as a guide but nota s a strict revert).

File: MouseUnSnag/Win32Interop/Win32Mouse.cs [hunk 1]
```diff
@@ -1,10 +1,6 @@
 ﻿using System;
-using System.Collections.Generic;
 using System.Drawing;
-using System.Linq;
 using System.Runtime.InteropServices;
-using System.Text;
-using System.Threading.Tasks;
```
Comment (patricknelson) [inline MouseUnSnag/Win32Interop/Win32Mouse.cs:7]: Cleanup: Just removing unused `using` statements.

**Comment by patricknelson** [comment]
Just setting the v2.2 milestone since the `Exit` capability currently in `master` is broken, so ideally either this or #5 should be merged prior to the next release.

**Comment by GrantMoyer** [comment]
> That reminds me: What’s the git workflow here? I’m assuming
> 
> * `master` branch is blessed (some use it as a dev or bleeding edge branch)
> * All feature branches are branched off of `master`
> * A `dev` branch may exist as a bleeding edge development branch (for integration)
> 
> 
> Only minor issue with that last one is just that the `dev` branch would have to be reverted or reset from time to time in case features don’t ever make it. In my projects, the `dev` branch doesn’t ever make it into `master` as it’s purely used for experimentation and ensuring feature branches work together. Thoughts?

I used dev branch to refer to what you call a feature branch (as in feature development branch). I agree that maintaining a separate, long lived branch for development doesn't make much sense for this repo. I think MouseUnSnag/MouseUnSnag really only needs a master branch, which gets tagged with releases. Most feature branches will be managed by developers in their own forks. In other words, it seems like I agree with your first two bullets, and share similar concerns with the third.

> Also, regarding the functionality itself, my approach would be:
> 
> * Remove it from `master` (the branch upon which we base our releases). My goal is to hopefully not only fix this `Exit` bug but to also get folks using the settings retention feature (which is super useful, see [Document settings retention (in snagConfig.txt) #11](https://github.com/MouseUnSnag/MouseUnSnag/issues/11)).
> * At _most_ convert it to a feature branch (i.e. branches off `master` and _only_ contains that additional feature). Only since we don't currently have a process for `dev` branch feature integration/testing (yet).

Yeah, I agree with this. I think we can skip converting it into a feature branch for now, until CSchwang chimes in, if ever.

**Comment by patricknelson** [comment]
Would love if you had a moment to review this @GrantMoyer, just to make sure I didn't miss or break anything. 😄 

Should get us a step closer to 2.2 (see https://github.com/MouseUnSnag/MouseUnSnag/milestone/1)

**Review by GrantMoyer — APPROVED** [review]
Looks good to me

**Comment by patricknelson** [comment]
Cool, thank you @GrantMoyer! Just need to do some documentation PR's and I think [v2.2](https://github.com/MouseUnSnag/MouseUnSnag/milestone/1) will be ready.

**Pull request merged**